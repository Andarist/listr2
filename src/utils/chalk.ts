import chalkDefault from 'chalk'

let chalk: chalkDefault.Chalk

/* istanbul ignore if */
if (process.env?.LISTR_DISABLE_COLOR !== '1' && chalkDefault.supportsColor) {
  chalk = new chalkDefault.Instance()
} else {
  chalk = new chalkDefault.Instance({ level: 0 })
}

/** A custom chalk instance to disable coloring compeletely. */
export default chalk
