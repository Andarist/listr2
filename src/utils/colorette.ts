import { createColors } from 'colorette'

export default createColors({ useColor: process.env?.LISTR_DISABLE_COLOR !== '1' })
