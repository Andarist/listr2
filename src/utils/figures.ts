import { isUnicodeSupported } from './is-unicode-supported'

const FIGURES_MAIN = {
  warning: '⚠',
  cross: '✖',
  arrowDown: '↓',
  tick: '✔',
  arrowRight: '→',
  pointer: '❯',
  checkboxOn: '☒',
  arrowLeft: '←',
  squareSmallFilled: '◼',
  pointerSmall: '›'
}

const FIGURES_FALLBACK = {
  ...FIGURES_MAIN,
  warning: '‼',
  cross: '×',
  tick: '√',
  pointer: '>',
  checkboxOn: '[×]',
  squareSmallFilled: '■'
}

export const figures = isUnicodeSupported() ? FIGURES_MAIN : FIGURES_FALLBACK
