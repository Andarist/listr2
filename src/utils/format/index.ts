export * from './colorette'
export * from './indent-string'
export * from './figures'
export * from './parse-time'
export * from './splat'
export * from './timestamp'
