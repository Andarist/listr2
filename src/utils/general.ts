import * as clone from 'clone'

/**
 * Deep clones a object in the most easiest manner.
 */
export function cloneObject<T extends Record<PropertyKey, any>> (obj: T): T {
  return clone(obj)
}
