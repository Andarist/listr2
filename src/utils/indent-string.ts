export function indentString (string: string, count: number): string {
  return string.replace(/^(?!\s*$)/gm, ' '.repeat(count))
}