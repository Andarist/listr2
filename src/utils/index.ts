export * from './environment'
export * from './format'
export * from './logger'
export * from './process-output'
export * from './ui'
export * from './assert'
export * from './clone'
export * from './concurrency'
export * from './delay'
export * from './uuid'
