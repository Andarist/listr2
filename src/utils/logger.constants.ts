/** Default loglevels for the logger */
export enum LogLevels {
  silent = 'silent',
  fail = 'fail',
  skip = 'skip',
  success = 'success',
  data = 'data',
  start = 'start',
  title = 'title',
  retry = 'retry',
  rollback = 'rollback'
}
