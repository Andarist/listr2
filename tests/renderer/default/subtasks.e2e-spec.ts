/* eslint-disable @typescript-eslint/no-empty-function */
import type { MockProcessOutput } from '@tests/utils'
import { delay, expectProcessOutputToMatchSnapshot, mockProcessOutput, unmockProcessOutput } from '@tests/utils'

import { Listr } from '@root'

describe('show subtasks', () => {
  const output: MockProcessOutput = {} as MockProcessOutput

  process.stdout.isTTY = true

  beforeEach(async () => {
    mockProcessOutput(output)
  })

  afterEach(async () => {
    unmockProcessOutput(output)
    jest.clearAllMocks()
  })

  // JQYWVb3x1scokQK2IzhwA4F0qxWbYzXR
  it('should create a set of subtasks', async () => {
    await new Listr(
      [
        {
          title: 'This task will execute.',
          task: (_, task): Listr =>
            task.newListr([
              {
                title: 'This is a subtask.',
                task: async (): Promise<void> => {}
              },

              {
                title: 'This is a subtask.',
                task: async (): Promise<void> => {}
              }
            ])
        }
      ],
      {
        concurrent: false,
        rendererOptions: { lazy: true }
      }
    ).run()

    expectProcessOutputToMatchSnapshot(output, 'JQYWVb3x1scokQK2IzhwA4F0qxWbYzXR')
  })

  // 3ygUxrTkCDLDfDrXzF1ocWR6626jaL0k
  it('should be able to change concurrency setting in subtask, second subtask should finish first', async () => {
    await new Listr(
      [
        {
          title: 'This task will execute.',
          task: (_, task): Listr =>
            task.newListr(
              [
                {
                  title: 'This is a subtask.',
                  task: async (): Promise<void> => {
                    await delay(10)
                  }
                },

                {
                  title: 'This is an another subtask.',
                  task: async (): Promise<void> => {}
                }
              ],
              { concurrent: true }
            )
        }
      ],
      {
        concurrent: false,
        rendererOptions: { lazy: true }
      }
    ).run()

    expectProcessOutputToMatchSnapshot(output, '3ygUxrTkCDLDfDrXzF1ocWR6626jaL0k')
  })

  // 8TNRrm8bI9ndz4jrhYC492luGNhtMTmZ
  it('should be able to change collapse settings', async () => {
    await new Listr(
      [
        {
          title: 'This task will execute.',
          task: (_, task): Listr =>
            task.newListr([
              {
                title: 'This is a subtask.',
                task: async (): Promise<void> => {}
              },

              {
                title: 'This is an another subtask.',
                task: async (): Promise<void> => {}
              }
            ])
        },

        {
          title: 'This task will execute.',
          task: (_, task): Listr =>
            task.newListr(
              [
                {
                  title: 'This is a subtask.',
                  task: async (): Promise<void> => {}
                },

                {
                  title: 'This is an another subtask.',
                  task: async (): Promise<void> => {}
                }
              ],
              { rendererOptions: { collapse: false } }
            )
        }
      ],
      {
        concurrent: false,
        rendererOptions: { lazy: true, collapse: true }
      }
    ).run()

    expectProcessOutputToMatchSnapshot(output, '8TNRrm8bI9ndz4jrhYC492luGNhtMTmZ')
  })

  // 12ttDlnth5pr1YuJEnfXm3I2CzRbcFlY
  it('should be able to change exit on error', async () => {
    try {
      await new Listr(
        [
          {
            title: 'This task will execute and not quit on errors.',
            task: (_, task): Listr =>
              task.newListr(
                [
                  {
                    title: 'This is a subtask.',
                    task: async (): Promise<void> => {
                      throw new Error('I have failed [0]')
                    }
                  },
                  {
                    title: 'This is an another subtask.',
                    task: async (): Promise<void> => {
                      throw new Error('I have failed [1]')
                    }
                  },
                  {
                    title: 'This is yet an another subtask.',
                    task: async (_, task): Promise<void> => {
                      task.title = 'I have succeeded.'
                    }
                  }
                ],
                { exitOnError: false }
              )
          },
          {
            title: 'This task will execute.',
            task: (): void => {
              throw new Error('exit')
            }
          }
        ],
        {
          concurrent: false,
          exitOnError: true,
          rendererOptions: { lazy: true }
        }
      ).run()
    } catch (e: any) {
      expect(e).toBeTruthy()
    }

    expectProcessOutputToMatchSnapshot(output, '12ttDlnth5pr1YuJEnfXm3I2CzRbcFlY')
  })

  // 9VMxFnaAKNjetcT1a9gLgN0dnasB7BRc
  it('should be able to render subtasks with no title with correct indentation', async () => {
    try {
      await new Listr(
        [
          {
            title: 'This task will execute.',
            task: (_, task): void => {
              task.output = 'output'
              task.output = 'some more output'
              task.output = 'even more output'
            },
            options: {
              bottomBar: Infinity,
              persistentOutput: true
            }
          },

          {
            title: 'This task will execute a task without titles.',
            task: (_, task): Listr =>
              task.newListr(
                [
                  {
                    task: async (): Promise<void> => {}
                  },
                  {
                    task: async (): Promise<void> => {}
                  },
                  {
                    task: async (): Promise<void> => {}
                  }
                ],
                { exitOnError: false }
              )
          }
        ],
        {
          concurrent: false,
          exitOnError: true,
          rendererOptions: { lazy: true }
        }
      ).run()
    } catch (e: any) {
      expect(e).toBeTruthy()
    }

    expectProcessOutputToMatchSnapshot(output, '9VMxFnaAKNjetcT1a9gLgN0dnasB7BRc')
  })

  // oGL7rWlMEOEDNqOOFgj2R9ilvahlLAuR
  it.each([ [ true, false ] ])('should output the task error in parent task with collapseErrors: %s and show subtasks: false', async (cases) => {
    try {
      await new Listr(
        [
          {
            title: 'This task will execute.',
            task: (_, task): Listr =>
              task.newListr([
                {
                  title: 'This is a subtask.',
                  task: async (): Promise<void> => {
                    throw new Error('bye')
                  }
                },

                {
                  title: 'This is a subtask.',
                  task: async (): Promise<void> => {}
                }
              ])
          }
        ],
        {
          concurrent: true,
          exitOnError: true,
          rendererOptions: {
            lazy: true,
            showSubtasks: false,
            collapseErrors: cases
          }
        }
      ).run()
    } catch (e: any) {
      expect(e).toBeTruthy()
    }

    expectProcessOutputToMatchSnapshot(output, 'oGL7rWlMEOEDNqOOFgj2R9ilvahlLAuR')
  })

  // bcaTpU40igebiNC4Koho5ObSuJxhYPsJ
  it.each([ true, false ])('should output the task error in parent task with: collapseErrors: %s showsubtasks: true', async (cases) => {
    try {
      await new Listr(
        [
          {
            title: 'This task will execute.',
            task: (_, task): Listr =>
              task.newListr([
                {
                  title: 'This is a subtask.',
                  task: async (): Promise<void> => {
                    throw new Error('bye')
                  }
                },

                {
                  title: 'This is a subtask.',
                  task: async (): Promise<void> => {}
                }
              ])
          }
        ],
        {
          concurrent: true,
          exitOnError: true,
          rendererOptions: {
            lazy: true,
            showSubtasks: true,
            collapseErrors: cases
          }
        }
      ).run()
    } catch (e: any) {
      expect(e).toBeTruthy()
    }

    expectProcessOutputToMatchSnapshot(output, 'bcaTpU40igebiNC4Koho5ObSuJxhYPsJ')
  })
})
