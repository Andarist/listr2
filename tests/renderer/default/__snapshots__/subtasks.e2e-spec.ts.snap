// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show subtasks should be able to change collapse settings: 8TNRrm8bI9ndz4jrhYC492luGNhtMTmZ-exit 1`] = `[]`;

exports[`show subtasks should be able to change collapse settings: 8TNRrm8bI9ndz4jrhYC492luGNhtMTmZ-stderr 1`] = `[]`;

exports[`show subtasks should be able to change collapse settings: 8TNRrm8bI9ndz4jrhYC492luGNhtMTmZ-stdout 1`] = `
[
  "‚óº This task will execute.
",
  "[2K[1A[2K[G‚óº This task will execute.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚óº This is a subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚óº This is a subtask.
  ‚óº This is an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚ùØ This is a subtask.
  ‚óº This is an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚úî This is a subtask.
  ‚óº This is an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚úî This is a subtask.
  ‚ùØ This is an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚úî This is a subtask.
  ‚úî This is an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚ùØ This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚ùØ This task will execute.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚ùØ This task will execute.
  ‚óº This is a subtask.
  ‚óº This is an another subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚ùØ This task will execute.
  ‚ùØ This is a subtask.
  ‚óº This is an another subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚ùØ This task will execute.
  ‚úî This is a subtask.
  ‚óº This is an another subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚ùØ This task will execute.
  ‚úî This is a subtask.
  ‚ùØ This is an another subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚ùØ This task will execute.
  ‚úî This is a subtask.
  ‚úî This is an another subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚úî This task will execute.
  ‚úî This is a subtask.
  ‚úî This is an another subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G",
  "‚úî This task will execute.
‚úî This task will execute.
  ‚úî This is a subtask.
  ‚úî This is an another subtask.
",
]
`;

exports[`show subtasks should be able to change concurrency setting in subtask, second subtask should finish first: 3ygUxrTkCDLDfDrXzF1ocWR6626jaL0k-exit 1`] = `[]`;

exports[`show subtasks should be able to change concurrency setting in subtask, second subtask should finish first: 3ygUxrTkCDLDfDrXzF1ocWR6626jaL0k-stderr 1`] = `[]`;

exports[`show subtasks should be able to change concurrency setting in subtask, second subtask should finish first: 3ygUxrTkCDLDfDrXzF1ocWR6626jaL0k-stdout 1`] = `
[
  "‚óº This task will execute.
",
  "[2K[1A[2K[G‚ùØ This task will execute.
",
  "[2K[1A[2K[G‚ùØ This task will execute.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚óº This is a subtask.
  ‚óº This is an another subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚ùØ This is a subtask.
  ‚óº This is an another subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚ùØ This is a subtask.
  ‚ùØ This is an another subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚ùØ This is a subtask.
  ‚úî This is an another subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚úî This is a subtask.
  ‚úî This is an another subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
",
  "[2K[1A[2K[G",
  "‚úî This task will execute.
",
]
`;

exports[`show subtasks should be able to change exit on error: 12ttDlnth5pr1YuJEnfXm3I2CzRbcFlY-exit 1`] = `[]`;

exports[`show subtasks should be able to change exit on error: 12ttDlnth5pr1YuJEnfXm3I2CzRbcFlY-stderr 1`] = `[]`;

exports[`show subtasks should be able to change exit on error: 12ttDlnth5pr1YuJEnfXm3I2CzRbcFlY-stdout 1`] = `
[
  "‚óº This task will execute and not quit on errors.
",
  "[2K[1A[2K[G‚óº This task will execute and not quit on errors.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚óº This is a subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚óº This is a subtask.
  ‚óº This is an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚óº This is a subtask.
  ‚óº This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚ùØ This is a subtask.
  ‚óº This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ This is a subtask.
  ‚óº This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚óº This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚ùØ This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚ùØ This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚ùØ I have succeeded.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ö† This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ö† This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚ùØ This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ö† This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚úñ This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ö† This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚úñ exit
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G",
  "‚ö† This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚úñ exit
",
]
`;

exports[`show subtasks should be able to render subtasks with no title with correct indentation: 9VMxFnaAKNjetcT1a9gLgN0dnasB7BRc-exit 1`] = `[]`;

exports[`show subtasks should be able to render subtasks with no title with correct indentation: 9VMxFnaAKNjetcT1a9gLgN0dnasB7BRc-stderr 1`] = `[]`;

exports[`show subtasks should be able to render subtasks with no title with correct indentation: 9VMxFnaAKNjetcT1a9gLgN0dnasB7BRc-stdout 1`] = `
[
  "‚óº This task will execute.
",
  "[2K[1A[2K[G‚óº This task will execute.
‚óº This task will execute a task without titles.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
‚óº This task will execute a task without titles.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
‚óº This task will execute a task without titles.

‚Ä∫ output
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
‚óº This task will execute a task without titles.

‚Ä∫ output
‚Ä∫ some more output
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
‚óº This task will execute a task without titles.

‚Ä∫ output
‚Ä∫ some more output
‚Ä∫ even more output
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚óº This task will execute a task without titles.

‚Ä∫ output
‚Ä∫ some more output
‚Ä∫ even more output
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚ùØ This task will execute a task without titles.

‚Ä∫ output
‚Ä∫ some more output
‚Ä∫ even more output
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
‚úî This task will execute a task without titles.

‚Ä∫ output
‚Ä∫ some more output
‚Ä∫ even more output
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G",
  "‚úî This task will execute.
‚úî This task will execute a task without titles.

‚Ä∫ output
‚Ä∫ some more output
‚Ä∫ even more output
",
]
`;

exports[`show subtasks should create a set of subtasks: JQYWVb3x1scokQK2IzhwA4F0qxWbYzXR-exit 1`] = `[]`;

exports[`show subtasks should create a set of subtasks: JQYWVb3x1scokQK2IzhwA4F0qxWbYzXR-stderr 1`] = `[]`;

exports[`show subtasks should create a set of subtasks: JQYWVb3x1scokQK2IzhwA4F0qxWbYzXR-stdout 1`] = `
[
  "‚óº This task will execute.
",
  "[2K[1A[2K[G‚ùØ This task will execute.
",
  "[2K[1A[2K[G‚ùØ This task will execute.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚óº This is a subtask.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚ùØ This is a subtask.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚úî This is a subtask.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚úî This is a subtask.
  ‚ùØ This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚úî This is a subtask.
  ‚úî This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚úî This task will execute.
",
  "[2K[1A[2K[G",
  "‚úî This task will execute.
",
]
`;

exports[`show subtasks should output the task error in parent task with collapseErrors: true and show subtasks: false: oGL7rWlMEOEDNqOOFgj2R9ilvahlLAuR-exit 1`] = `[]`;

exports[`show subtasks should output the task error in parent task with collapseErrors: true and show subtasks: false: oGL7rWlMEOEDNqOOFgj2R9ilvahlLAuR-stderr 1`] = `[]`;

exports[`show subtasks should output the task error in parent task with collapseErrors: true and show subtasks: false: oGL7rWlMEOEDNqOOFgj2R9ilvahlLAuR-stdout 1`] = `
[
  "‚óº This task will execute.
",
  "[2K[1A[2K[G‚ùØ This task will execute.
",
  "[2K[1A[2K[G",
  "‚ùØ This task will execute.
",
]
`;

exports[`show subtasks should output the task error in parent task with: collapseErrors: false showsubtasks: true: bcaTpU40igebiNC4Koho5ObSuJxhYPsJ-exit 1`] = `[]`;

exports[`show subtasks should output the task error in parent task with: collapseErrors: false showsubtasks: true: bcaTpU40igebiNC4Koho5ObSuJxhYPsJ-stderr 1`] = `[]`;

exports[`show subtasks should output the task error in parent task with: collapseErrors: false showsubtasks: true: bcaTpU40igebiNC4Koho5ObSuJxhYPsJ-stdout 1`] = `
[
  "‚óº This task will execute.
",
  "[2K[1A[2K[G‚ùØ This task will execute.
",
  "[2K[1A[2K[G‚ùØ This task will execute.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚óº This is a subtask.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚ùØ This is a subtask.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚úñ This is a subtask.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚úñ This is a subtask.
    ‚Ä∫ bye
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G",
  "‚ùØ This task will execute.
  ‚úñ This is a subtask.
    ‚Ä∫ bye
  ‚óº This is a subtask.
",
]
`;

exports[`show subtasks should output the task error in parent task with: collapseErrors: true showsubtasks: true: bcaTpU40igebiNC4Koho5ObSuJxhYPsJ-exit 1`] = `[]`;

exports[`show subtasks should output the task error in parent task with: collapseErrors: true showsubtasks: true: bcaTpU40igebiNC4Koho5ObSuJxhYPsJ-stderr 1`] = `[]`;

exports[`show subtasks should output the task error in parent task with: collapseErrors: true showsubtasks: true: bcaTpU40igebiNC4Koho5ObSuJxhYPsJ-stdout 1`] = `
[
  "‚óº This task will execute.
",
  "[2K[1A[2K[G‚ùØ This task will execute.
",
  "[2K[1A[2K[G‚ùØ This task will execute.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚óº This is a subtask.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚ùØ This is a subtask.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚úñ This is a subtask.
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute.
  ‚úñ bye
  ‚óº This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G",
  "‚ùØ This task will execute.
  ‚úñ bye
  ‚óº This is a subtask.
",
]
`;
